extension/src/extension.ts

```diff
@@ -1,13 +1,97 @@
 'use strict';
 
 import * as vscode from 'vscode';
+import { AuthorizationRequest } from "@openid/appauth/built/authorization_request";
+import { AuthorizationServiceConfiguration } from "@openid/appauth/built/authorization_service_configuration";
+import { NodeBasedHandler } from "@openid/appauth/built/node_support/node_request_handler";
+import { NodeRequestor } from "@openid/appauth/built/node_support/node_requestor";
+import { AuthorizationNotifier } from "@openid/appauth/built/authorization_request_handler";
+import {
+    GRANT_TYPE_AUTHORIZATION_CODE,
+    TokenRequest
+} from "@openid/appauth/built/token_request";
+import { BaseTokenRequestHandler } from "@openid/appauth/built/token_request_handler";
+import { TokenResponse } from '@openid/appauth/built/token_response';
 
 export function activate(context: vscode.ExtensionContext) {
+    context.subscriptions.push(vscode.commands.registerCommand('extension.signIn', async () => await signIn()));
+}
+
+export function deactivate() {
+}
+
+let clientId: string = 'sample-openId-extension';
+let authPort: number = 58805;
+let redirectUri = 'http://localhost:' + authPort;
+let state: string;
+
+function signIn(){
+    let configuration: AuthorizationServiceConfiguration;
+    let nodeRequester = new NodeRequestor();
```

Comment:
nit: the requester/requestor thing bothers me - might as well be consistent, I think?